tooling/debugger/src/context.rs

```diff
@@ -142,6 +185,47 @@ impl<'a, B: BlackBoxFunctionSolver> DebugContext<'a, B> {
         }
     }
 
+    fn breakpoint_reached(&self) -> bool {
+        if let Some(location) = self.get_current_opcode_location() {
+            self.breakpoints.contains(&location)
+        } else {
+            false
+        }
+    }
+
+    pub(super) fn is_valid_location(&self, location: &OpcodeLocation) -> bool {
+        let opcodes = self.get_opcodes();
+        match *location {
+            OpcodeLocation::Acir(acir_index) => acir_index < opcodes.len(),
+            OpcodeLocation::Brillig { acir_index, brillig_index } => {
+                acir_index < opcodes.len()
+                    && matches!(opcodes[acir_index], Opcode::Brillig(..))
+                    && {
+                        let Opcode::Brillig(ref brillig) = opcodes[acir_index] else {
+                            unreachable!("opcode at {acir_index} is not Brillig")
+                        };
+                        brillig_index < brillig.bytecode.len()
```

Comment:
```suggestion
                        if let Some(Opcode::Brillig(ref brillig)) = opcodes[acir_index] {
                            brillig_index < brillig.bytecode.len()
                        } else {
                            false
                        }
```

There's no need to panic here.